Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.BindingsCollection.cs

```diff
@@ -0,0 +1,105 @@
+// CodeContracts
+// 
+// Copyright (c) Microsoft Corporation
+// 
+// All rights reserved. 
+// 
+// MIT License
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Drawing;
+using System.Diagnostics.Contracts;
+using System.Windows.Forms.Layout;
+using System.ComponentModel;
+
+namespace System.Windows.Forms
+{
+  /// <summary>
+  /// Represents a collection of <see cref="T:System.Windows.Forms.Binding"/> objects for a control.
+  /// </summary>
+  public class BindingsCollection : BaseCollection
+  {
+    /// <summary>
+    /// Gets the total number of bindings in the collection.
+    /// </summary>
+    /// 
+    /// <returns>
+    /// The total number of bindings in the collection.
+    /// </returns>  
+    public override int Count
+    {
+        get
+        {
+            Contract.Ensures(Contract.Result<int>() > 0);
```

Comment:
I can't see a reason why we need to add another postcondition to overriden method.

Both static checker and rewriter will use postcondition from base implementation.

Derived class theoretically can provide stronger postconditions, but I don't think that CC support this. But there is no need to duplicate the code, especially because this postcondition looks suspicions to me (I think it should be `Contract.Result<int>() >= 0` instead).